os/scripts/gen-keystone-kustomize.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+#
+# Copyright 2022 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+set -ex
+
+# expect that the common.sh is in the same dir as the calling script
+SCRIPTPATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
+. ${SCRIPTPATH}/common.sh --source-only
+
+NAMESPACE=$1
+KEYSTONE_ADMIN_PWD=$2
+KEYSTONE_DB_PWD=$3
+
+if [ -z "$NAMESPACE" ]; then
+      echo "Please set NAMESPACE as ARG1"; exit 1
+fi
+
+if [ -z "$KEYSTONE_ADMIN_PWD" ]; then
+      echo "Please set KEYSTONE_ADMIN_PWD as ARG2"; exit 1
+fi
+
+if [ -z "$KEYSTONE_DB_PWD" ]; then
+      echo "Please set KEYSTONE_DB_PWD as ARG3"; exit 1
+fi
+
+CR_DIR=out/${NAMESPACE}/keystone/cr
+
+if [ ! -d ${CR_DIR} ]; then
+      mkdir -p ${CR_DIR}
+fi
+
+pushd ${CR_DIR}
+
+cat <<EOF >kustomization.yaml
```

Comment:
These files are quite verbose. I've been using a combined k8s secret for devel (essentially sharing one secret for multiple services). Not recommended for production of course but having a HERE doc that spits out one k8s secret for all the things is much simpler for dev environments.

And with regards to kustomize I think the way I was thinking about using it was more of a we have a set of defaults everyone uses. But if I want to have my own defaults I create my own branch in my fork, and can go do whatever I want and layer over things with kustimize. Things like custom images, or whatever. That way we don't have to maintain shell scripts with options for all the things. Just a thought on the pattern we develop for this. Essentially trying to keep the set of scripts to something quite minimal

Reply:
yes we could do that. I did something similar to create multiple secrets, but for the password they e.g default all to the same password passed in via the Makefile. I have not checked the cnosp operators to what they expect to be in that secret. we might have to extend that this single secret holds all the information a controller expects. like rabbit, passwords to connect to another OSP service. 

I was thinking for the CMs/Secrets the operators require for input, that it is probably better to use labels to get the object instead of the CM/secret name itself. with this its possible to have a combined object holding information for multiple controllers.